web/packages/teleterm/src/services/tshd/errors.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright 2023 Gravitational, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export function isAccessDeniedError(error: Error): boolean {
+  return (error.message as string)?.includes('access denied');
```

Comment:
It's going to be a mess once we remove the following piece of code: https://github.com/gravitational/teleport/blob/29211292a0e3a029ef60eac93d13f06dc7bab525/lib/teleterm/apiserver/middleware.go#L39-L41

But at the moment there's no way around it. I suppose at some point we'll have to implement #30753 and then we'll be able to base this on a response code instead.